src/main/cpp/util/port.h

```diff
@@ -71,6 +71,21 @@
 #endif
 
 
+// CAN_FIND_OWN_EXECUTABLE_PATH
+//
+// Indicates that a running process can find a path to its own executable.
+#if !defined(__OpenBSD__)
```

Comment:
This logic assumes every other platform can find their own executable. That isn't guaranteed. What I believe you actually want is:
```
#if defined(__OpenBSD__)
#define CANNOT_FIND_OWN_EXECUTABLE_PATH=1
#endif
```

This tells a different story: we know on OpenBSD we can't find our own executable, without assuming anything of other platforms.

Reply:
That would also work. Before I make this change, though: Are you sure?

- The concern is that if we use `CAN_FIND_OWN_EXECUTABLE_PATH` and forget to update `port.h` when adding a new platform, we will silently assume that the new platform supports binaries finding their own executable paths. But, the same is equally true of the `CANNOT_FIND_OWN_EXECUTABLE_PATH` alternative: If we forget to update `port.h` when adding a new platform, then `CANNOT_FIND_OWN_EXECUTABLE_PATH` won't be `#define`d for the new platform, and thus the rest of the code will assume that the platform supports binaries finding their own executable paths.

- The negative naming is hard to read. Code is easier to read when symbols say what *is* true, not what *isn't* true. For example, we probably wouldn't want to refactor `HAVE_ATTRIBUTE` as `DONT_HAVE_ATTRIBUTE`.

So, AFAICS the proposed change would not meet the stated goal, and meanwhile would make the code less readable.

A robust (but longer) solution might be something like this:

```
#if defined(__linux) || defined(__APPLE__) || __defined(__FreeBSD__) || defined(_WIN32) || defined(__CYGWIN__)
#define CAN_FIND_OWN_EXECUTABLE_PATH
#elif defined(__OpenBSD__)
// This platform does not support processes finding their own executable paths.
#else
#error Unrecognized platform.
#endif
```

What do you think?

Reply:
Thanks for explaining that. You're right, my solution also makes assumptions about other platforms.

The robust solution looks safer than the other two approaches without much more complexity, so yes I think that looks like the best solution.

Reply:
Done.